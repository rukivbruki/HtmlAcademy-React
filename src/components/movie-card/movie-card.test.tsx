import * as React from "react";
import * as renderer from "react-test-renderer";
import {MemoryRouter} from 'react-router';
import MovieCard from './movie-card';

const props = {
  key: 1,
  id: 1,
  name: ``,
  genre: ``,
  previewImage: ``,
  videoLink: ``,
};

it(`MovieCard renders correctly`, () => {
  const tree = renderer
    .create(
        <MemoryRouter initialEntries={[`/test`]} >
          <MovieCard {...props}/>
        </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
