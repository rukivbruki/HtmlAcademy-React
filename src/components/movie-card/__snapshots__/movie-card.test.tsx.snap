// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt=""
      height="175"
      src=""
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "display": "flex",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      
    </a>
  </h3>
</article>
`;
